File: java/testsuite/Jogl4Examples/MeshExample/src/gui/MeshMouseInteractionTechniques.java
package gui;

import vsdk.toolkit.gui.MouseEvent;
import vsdk.toolkit.gui.CameraController;

public class MeshMouseInteractionTechniques {
    private final CameraController cameraController;

    public MeshMouseInteractionTechniques(CameraController cameraController)
    {
        this.cameraController = cameraController;
    }

    public boolean processMousePressedEvent(MouseEvent event)
    {
        return cameraController.processMousePressedEvent(event);
    }

    public boolean processMouseReleasedEvent(MouseEvent event)
    {
        return cameraController.processMouseReleasedEvent(event);
    }

    public boolean processMouseClickedEvent(MouseEvent event)
    {
        return cameraController.processMouseClickedEvent(event);
    }

    public boolean processMouseMovedEvent(MouseEvent event)
    {
        return cameraController.processMouseMovedEvent(event);
    }

    public boolean processMouseDraggedEvent(MouseEvent event)
    {
        return cameraController.processMouseDraggedEvent(event);
    }

    public boolean processMouseWheelEvent(MouseEvent event)
    {
        System.out.println(".");
        return cameraController.processMouseWheelEvent(event);
    }
}

File: cpp/testsuite/OpenGL4Examples/MeshExample/src/gui/MeshMouseInteractionTechniques.cpp
#include <cstdio>

#include "vsdk/toolkit/gui/CameraController.h"
#include "gui/MeshMouseInteractionTechniques.h"

MeshMouseInteractionTechniques::MeshMouseInteractionTechniques(CameraController* cameraController)
    : cameraController(cameraController)
{
}

bool MeshMouseInteractionTechniques::processMousePressedEvent(const MouseEvent& event)
{
    return cameraController != 0 && cameraController->processMousePressedEvent(event);
}

bool MeshMouseInteractionTechniques::processMouseReleasedEvent(const MouseEvent& event)
{
    return cameraController != 0 && cameraController->processMouseReleasedEvent(event);
}

bool MeshMouseInteractionTechniques::processMouseClickedEvent(const MouseEvent& event)
{
    return cameraController != 0 && cameraController->processMouseClickedEvent(event);
}

bool MeshMouseInteractionTechniques::processMouseMovedEvent(const MouseEvent& event)
{
    return cameraController != 0 && cameraController->processMouseMovedEvent(event);
}

bool MeshMouseInteractionTechniques::processMouseDraggedEvent(const MouseEvent& event)
{
    return cameraController != 0 && cameraController->processMouseDraggedEvent(event);
}

bool MeshMouseInteractionTechniques::processMouseWheelEvent(const MouseEvent& event)
{
    std::printf(".\n");
    return cameraController != 0 && cameraController->processMouseWheelEvent(event);
}
